### PR TITLE
OCPBUGS-6945: Fixes node OS detection

### PR DESCRIPTION
**Pull request by cheesesashimi**
**- What I did**

The OS release node detection was not correctly detecting RHCOS 9 OSes due to the VERSION_ID field being populated with the OCP version instead of the OS version.

**- How to verify it**

1. Run the unit test suite.
2. Deploy an OCP cluster running either RHCOS 8 or RHCOS 9 (for RHCOS 9, provision a cluster using ClusterBot pointed at https://github.com/openshift/machine-config-operator/pull/3485).
3. Run the e2e test suite. The e2e test will output additional log entries indicating the node OS and, in the case of RHCOS 8 or 9, the version it is running, which should match the `RHEL_VERSION` field in the `/etc/os-release` / `/usr/lib/os-release` files on the node.

These logs will resemble:
```console
go test -v -count=1 -run="TestOSDetection" .
=== RUN   TestOSDetection
=== RUN   TestOSDetection/#00
=== PAUSE TestOSDetection/#00
=== RUN   TestOSDetection/#01
=== PAUSE TestOSDetection/#01
=== RUN   TestOSDetection/#02
=== PAUSE TestOSDetection/#02
=== RUN   TestOSDetection/#03
=== PAUSE TestOSDetection/#03
=== RUN   TestOSDetection/#04
=== PAUSE TestOSDetection/#04
=== RUN   TestOSDetection/#05
=== PAUSE TestOSDetection/#05
=== CONT  TestOSDetection/#00
=== CONT  TestOSDetection/#03
=== CONT  TestOSDetection/#02
=== CONT  TestOSDetection/#04
=== CONT  TestOSDetection/#01
=== CONT  TestOSDetection/#05
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-173-134.ec2.internal with version 8.6
=== CONT  TestOSDetection/#04
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-163-214.ec2.internal with version 8.6
=== CONT  TestOSDetection/#01
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-138-110.ec2.internal with version 8.6
=== CONT  TestOSDetection/#00
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-134-197.ec2.internal with version 8.6
=== CONT  TestOSDetection/#02
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-147-135.ec2.internal with version 8.6
=== CONT  TestOSDetection/#03
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-152-25.ec2.internal with version 8.6
--- PASS: TestOSDetection (0.08s)
    --- PASS: TestOSDetection/#05 (0.67s)
    --- PASS: TestOSDetection/#04 (0.67s)
    --- PASS: TestOSDetection/#01 (0.70s)
    --- PASS: TestOSDetection/#00 (0.73s)
    --- PASS: TestOSDetection/#02 (0.74s)
    --- PASS: TestOSDetection/#03 (0.83s)
PASS
ok      github.com/openshift/machine-config-operator/test/e2e   1.566s
```

**- Description for the changelog**
Fixes node OS detection

### CODE REVIEW

**Comment by cgwalters** [comment]
/approve
/lgtm
Looks sane to me, though I'd say I think at some point we should do feature detection (e.g. `rpm -q openssh`) - that way it'd work automatically on Alma Linux too etc.

**Comment by cheesesashimi** [comment]
Agreed about the feature detection. I guess the big TODO there is determining which features we need and understanding how to determine whether those features are there (e.g., `rpm -q openssh`, `rpm-ostree version`, etc.).

**Comment by cheesesashimi** [comment]
Now that I've fixed a few issues with the unit tests, I'd like both @cgwalters and @LorbusChris to weigh in on this PR.

**Comment by rioliu-rh** [comment]
- setup cluster via clusterbot+openshift/machine-config-operator#3485
- check os-release file content on worker node
```
 oc debug node/ip-10-0-139-113.us-west-1.compute.internal -- chroot /host cat /etc/os-release
Temporary namespace openshift-debug-hgp9x is created for debugging node...
Starting pod/ip-10-0-139-113us-west-1computeinternal-debug ...
To use host binaries, run `chroot /host`
NAME="Red Hat Enterprise Linux CoreOS"
ID="rhcos"
ID_LIKE="rhel fedora"
VERSION="413.90.202212151724-0"
VERSION_ID="4.13"
VARIANT="CoreOS"
VARIANT_ID=coreos
PLATFORM_ID="platform:el9"
PRETTY_NAME="Red Hat Enterprise Linux CoreOS 413.90.202212151724-0 (Plow)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:9::coreos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://docs.openshift.com/container-platform/4.13/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="OpenShift Container Platform"
REDHAT_BUGZILLA_PRODUCT_VERSION="4.13"
REDHAT_SUPPORT_PRODUCT="OpenShift Container Platform"
REDHAT_SUPPORT_PRODUCT_VERSION="4.13"
OPENSHIFT_VERSION="4.13"
RHEL_VERSION="9.0"
OSTREE_VERSION="413.90.202212151724-0"

Removing debug pod ...
Temporary namespace openshift-debug-hgp9x was removed.
```
- run unit test, test output is like below
```
go test -v -count=1 -run="TestOSDetection" .
=== RUN   TestOSDetection
=== RUN   TestOSDetection/#00
=== PAUSE TestOSDetection/#00
=== RUN   TestOSDetection/#01
=== PAUSE TestOSDetection/#01
=== RUN   TestOSDetection/#02
=== PAUSE TestOSDetection/#02
=== RUN   TestOSDetection/#03
=== PAUSE TestOSDetection/#03
=== RUN   TestOSDetection/#04
=== PAUSE TestOSDetection/#04
=== CONT  TestOSDetection/#00
=== CONT  TestOSDetection/#03
=== CONT  TestOSDetection/#04
=== CONT  TestOSDetection/#02
=== CONT  TestOSDetection/#01
=== CONT  TestOSDetection/#00
    node_os_test.go:68: Identified Red Hat Enterprise Linux CoreOS 9.0 on node ip-10-0-139-113.us-west-1.compute.internal
=== CONT  TestOSDetection/#01
    node_os_test.go:68: Identified Red Hat Enterprise Linux CoreOS 9.0 on node ip-10-0-154-69.us-west-1.compute.internal
=== CONT  TestOSDetection/#04
    node_os_test.go:68: Identified Red Hat Enterprise Linux CoreOS 9.0 on node ip-10-0-220-74.us-west-1.compute.internal
=== CONT  TestOSDetection/#03
    node_os_test.go:68: Identified Red Hat Enterprise Linux CoreOS 9.0 on node ip-10-0-183-67.us-west-1.compute.internal
=== CONT  TestOSDetection/#02
    node_os_test.go:68: Identified Red Hat Enterprise Linux CoreOS 9.0 on node ip-10-0-179-181.us-west-1.compute.internal
--- PASS: TestOSDetection (1.12s)
    --- PASS: TestOSDetection/#00 (20.86s)
    --- PASS: TestOSDetection/#01 (21.44s)
    --- PASS: TestOSDetection/#04 (21.47s)
    --- PASS: TestOSDetection/#03 (21.62s)
    --- PASS: TestOSDetection/#02 (22.63s)
PASS
ok  	github.com/openshift/machine-config-operator/test/e2e	25.005s
```

**Comment by rioliu-rh** [comment]
label qe-approved

**Comment by rioliu-rh** [comment]
@cheesesashimi @cgwalters verified with unit test, please confirm, thanks

**Comment by cheesesashimi** [comment]
[This failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3529/pull-ci-openshift-machine-config-operator-master-e2e-aws-ovn/1620921660477542400) in `e2e-aws-ovn` is related to https://issues.redhat.com/browse/TRT-819. I am reasonably confident it was not caused by the changes in this PR.

File: pkg/daemon/osrelease/osrelease.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package osrelease
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ashcrow/osrelease"
```
Comment (cgwalters) [inline pkg/daemon/osrelease/osrelease.go:8]: This is fine as is, but perhaps it'd make sense at some point to have this functionality move into that upstream library.  We could also move it to e.g. github.com/coreos/osrelease too for more shared maintenance.

**Review by cgwalters — APPROVED** [review]
/lgtm

File: pkg/daemon/osrelease/osrelease.go (hunk 1 above)
Comment (cheesesashimi) [reply pkg/daemon/osrelease/osrelease.go:8]: Yeah, that would be nice. I have a few ideas around some additional things we could port there such as handling paths, etc.

**Pull request merged**